src/languages/regex.js

```diff
@@ -0,0 +1,101 @@
+/*
+Language: regex
+Description: Regular expressions 
+Author: Konrad Rudolph <konrad.rudolph@gmail.com>
+*/
+
+export default function (hljs) {
+  const ESCAPE_SEQUENCE = {
+    className: 'built_in',
+    begin: [
+      // Unicode script/block/category
+      /\\[pP](?:[^{]|\{\^?[^}]+\})/,
+      // Unicode character
+      /\\[ux](?:[0-9a-fA-F]{1,4}|\{[0-9a-fA-F]{1,4}\})/,
+      // Backreferences
+      /\\k(?:'-?\d+'|<-?\d+>|{-?\d+})/,
+      /\\g(?:'-?\d+'|<-?\d+>|{-?\d+}|-?\d+)/,
+      // Everything else
+      /\\./
+    ].map(r => r.source).join('|')
+  };
+
+  const CHARACTER_CLASS = {
+    className: 'string',
+    begin: /\[\^?]?/,
+    end: /]/,
+    contains: [
+      {
+        className: 'literal',
+        variants: [
+          // POSIX named character class.
+          { begin: /\[:\^?/, end: /:]/ },
+          // POSIX collation sequences.
+          { begin: /\[\./, end: /\.]/ },
+          // POSIX character equivalence.
+          { begin: /\[=/, end: /=]/ }
+        ]
+      },
+      ESCAPE_SEQUENCE
+    ]
+  };
+
+  const META = {
+    className: 'keyword',
+    begin: /[.+*?|^$]|\{(?:\d+(?:,\d*)?|,\d+)\}/,
+  };
+
+  const COMMENT = hljs.COMMENT(/\(\?#/, /\)/);
+
+  const MODIFIERS = '-?(?:[ictsmnpdJUbqX^]|xx?)';
+
+  const GROUP_SPECIALS = [
+    // Named group
+    'P?<[^>]+>',
+    "'[^']+'",
+    // Zero-length assertions, branch reset & atomic
+    '[|>=!]|<=|<!',
+    // Modifiers applied inside group
+    `(?:${MODIFIERS}):`,
+    ':'
+  ];
+
+  const MODIFIER = {
+    className: 'meta',
+    begin: `\\(\\?(?:${MODIFIERS})\\)`
```

Comment:
Not sure I know what is happening here... example? If they are literally modifiers then `regex.modifier` doesn't seem so terrible...

Reply:
Yeah, `regex.modifier` would be apt. I was going to add explanations via test cases but I haven’t been doing this systematically yet. Here’s an example:

For example, `((?i)foo)bar` matches `foo` in a case-insensitive manner, followed by `bar` case-sensitively. It’s the equivalent of regular modifiers (`/foobar/i`), except applied to a subexpression only.

Reply:
OMG, does that local `i` mod work in JS regex? :-)

Reply:
Doesn’t look like it. I usually refer to the (incredibly broken) www.regular-expressions.info. It has the most comprehensive cross-language reference I’m aware of, and also lists (albeit now outdated) details about the scope of each implementation, e.g. [for JavaScript](https://www.regular-expressions.info/javascript.html).

Reply:
Yeah, I couldn't get it working now trying both syntaxes...